### PR TITLE
Initial unit tests

### LINKED ISSUE #24: Pickle files to transform raw data has to be done only for used dataset
Right now, if .pkl files are already in side serialized_dataset, new .pkl files are not created. 
And if no .pkl files are inside the serialized_dataset folder, then ALL of them are created when the code is run for the first time.

### PR DESCRIPTION
**Pull request by allaffa**
Part of #33 

Fixes #24

### CODE REVIEW

**Comment by allaffa** [comment]
@streeve @pzhanggit  

I understand that something in the code need to be changed. But given the current structure of the code, I think all the model pass the tests with a random data generated from scratch. 
New changes can be suggested in a new pull request after this one has been merged.

**Review by pzhanggit — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 *~
 *DS_Store
 __pycache__
-logs
```
Comment (streeve) [inline .gitignore:4]: Can you keep ignoring this? Or did it break things?

File: test_random_graphs.py [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
 from run_config_input import run_normal_config_file
+from utils.random_graph_data import random_graph_data
+from test_trained_model import test_trained_model
 
 
 @pytest.mark.mpi()
 @pytest.mark.parametrize("model_type", ["GIN", "GAT", "MFC", "PNN"])
 def pytest_train_model(model_type):
+
+    comm = MPI.COMM_WORLD
```
Comment (streeve) [inline test_random_graphs.py:15]: Instead, just use `get_comm_size_and_rank` and remove `mpi4py`

File: test_trained_model.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
```
Comment (streeve) [inline test_trained_model.py:3]: I don't think we need the header (mostly this would be the only file that has it)

File: test_trained_model.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+Created on Tue Jul 27 19:43:21 2021
+
+@author: 7ml
+"""
+import sys, os, json
+import shutil
+
+import torch
+from torch.utils.tensorboard import SummaryWriter
+from torch.optim.lr_scheduler import ReduceLROnPlateau
```
Comment (streeve) [inline test_trained_model.py:13]: Remove

File: test_trained_model.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    if chosen_model is None:
+        raise RuntimeError("No model type provided")
+
+    run_in_parallel, world_size, world_rank = setup_ddp()
+
+    config = {}
+    with open(config_file, "r") as f:
+        config = json.load(f)
+    predicted_value_option = {1: 1, 2: 32, 3: 32, 4: 33, 5: 33, 6: 65}
```
Comment (streeve) [inline test_trained_model.py:41]: What is this?

File: utils/random_graph_data.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
```
Comment (streeve) [inline utils/random_graph_data.py:3]: Same comment about headers

File: test_trained_model.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    config["output_dim"] = predicted_value_option[config["predicted_value_option"]]
+
+    dataset_options = {
```
Comment (streeve) [inline test_trained_model.py:44]: This should be a separate function (but I know you would prefer to merge this and clean it up later)

**Review by streeve — CHANGES_REQUESTED** [review]
I see the unit_test pickle in the repo - can't that be deleted?

File: test_trained_model.py (hunk 7 above)
Comment (allaffa) [reply test_trained_model.py:44]: @streeve 

I can remove this now, but maybe Pei needs it for her additional multi-task uni tests. 
If we need to have a test_trained model, I would like to have one that we can use on every model possible. 
@pzhanggit what do you think?

**Comment by allaffa** [comment]
@streeve @jychoi-hpc 

I think that everything that has to do with creation, modification from raw data to .pkl data, and destruction of folders/data has to be safeguarded by barriers and per performed only by the master process.

File: test_trained_model.py (hunk 7 above)
Comment (streeve) [reply test_trained_model.py:44]: No, I wouldn't remove, just make a function like `get_dataset`

Comment (pzhanggit) [reply test_trained_model.py:44]: Are you suggesting to infer the output dimension from dataset? That also works for me.  As long as we do not limit the number of nodes/atoms to 32...

File: .gitignore (hunk 1 above)
Comment (streeve) [reply .gitignore:4]: Can you remove?

File: test_trained_model.py (hunk 7 above)
Comment (streeve) [reply test_trained_model.py:32]: We can just fix later

**Review by pzhanggit — APPROVED** [review]

**Review by streeve — APPROVED** [review]

**Pull request merged**